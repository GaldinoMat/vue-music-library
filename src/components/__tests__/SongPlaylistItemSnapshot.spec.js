import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import SongPlaylistItem from '@/components/SongPlaylistItem/SongPlaylistItem.vue'

const songFactory = () => {
  const song = {
    docId: 'abc',
    modifiedName: 'file_example_MP3_1MG.mp3',
    displayName: 'TestAuthor',
    commentCount: 5
  }

  return song
}

const wrapperFactory = (newSong) => {
  const wrapper = shallowMount(SongPlaylistItem, {
    props: {
      song: newSong
    },
    global: {
      components: {
        RouterLink: RouterLinkStub
      }
    }
  })

  return wrapper
}

describe('SongPlaylistItem snapshot', () => {
  it('should render correctly', () => {
    const song = songFactory()
    const wrapper = wrapperFactory(song)

    expect(wrapper.element).toMatchSnapshot()
  })
})
