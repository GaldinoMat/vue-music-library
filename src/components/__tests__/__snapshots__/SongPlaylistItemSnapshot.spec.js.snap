// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongPlaylistItem snapshot > should render correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
  id="songId-abc"
>
  <div>
    <router-link-stub
      ariacurrentvalue="page"
      class="font-bold block text-gray-600 composition-name"
      custom="false"
      id="SongName"
      replace="false"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
      id="SongAuthor"
    >
      TestAuthor
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      ariacurrentvalue="page"
      custom="true"
      replace="false"
      to="[object Object]"
    />
  </div>
</li>
`;
